utils_cv/action_recognition/model.py

```diff
@@ -61,16 +72,21 @@ def __init__(
             kinetics. By default it will use the weights from ig65m since it
             tends attain higher results.
         """
-        self.dataset = dataset
-        self.model, self.model_name = self.init_model(
-            self.dataset.sample_length, base_model, num_classes,
-        )
+        # set empty - populated when fit is called
+        self.results = []
 
         # set num classes
         self.num_classes = num_classes
 
-        # set empty - populated when fit is called
-        self.results = []
+        if dataset:
+            self.dataset = dataset
+            self.sample_length = self.dataset.sample_length
+        else:
+            self.sample_length = 8 if sample_length <= 8 else 32
```

Comment:
Maybe add an assert here to make sure the user specifies either 8 or 32, rather than overwriting user input silently

Reply:
done